src/main/zapHomeFiles/hud/tools/activeScan.js

```diff
@@ -87,12 +87,43 @@ var ActiveScan = (function() {
 			.catch(utils.errorHandler);
 	}
 
-	function startActiveScan(tabId, domain) {
+	function startActiveScanDomain(tabId, domain) {
 		utils.getUpgradedDomain(domain)
 			.then(upgradedDomain => {
-				return utils.zapApiCall("/ascan/action/scan/?url=" + upgradedDomain)
-			}).
-			then(response => {
+				startActiveScan(tabId, upgradedDomain, "true", "GET", "");
+			})
+			.catch(utils.errorHandler);
+	}
+	
+	function showReqDialog(tabId, uri, method, body) {
+		Promise.all([checkIsRunning(tabId), self.tools.scope.isInScope(uri)])
+			.then(results => {
+				var isRunning = results[0];
+				var isInScope = results[1];
+				var config = {};
+				config.buttons = [{text: I18n.t("common_cancel"), id: "cancel"}];
+
+				if (!isInScope) {
+					config.text = DIALOG.START_ADD_SCOPE;
+					config.buttons.unshift({text: I18n.t("common_start"), id: "start-add-to-scope"});
+				}
+				else {
+					config.text = DIALOG.START;
+					config.buttons.unshift({text: I18n.t("common_start"), id: "start"});
+				}
+				return config;
+			})
+			.then(config => {
+				return utils.messageFrame(tabId, "display", {action:"showDialog", config:config})
+			})
+			.catch(utils.errorHandler);
+	}
+
+
+	function startActiveScan(tabId, uri, recurse, method, body) {
```

Comment:
I'd prefer to keep this method as simple as needed. For example, do we have a need for changing the method? or adding a body? I think we do need to differentiate between scanning a url vs scanning a domain, but that could either be handled by two functions `startActiveScanDomain` vs `startActiveScanUri` and those should both only expose the parameters they need for now. Realistically though I think we could get away with a single function `startActiveScan(tabId, uri, recurse)` which would allow us to support both use cases in a single function with just the parameters we need.

Reply:
Yes we do need these - they are needed (and specified) when scanning a POST request. If they're not provided then we'll end up scanning a GET request instead, which isnt what we want to do.

Reply:
ah gotcha. I see it now. thanks!